contributors/design-proposals/storage/container-storage-interface-inline-volumes.md

```diff
@@ -82,34 +76,40 @@ type CSIVolumeSource struct {
 	// +optional
 	NodePublishSecretRef *LocalObjectReference
 }
-
-type CSIVolumeHandlePrefix string
-const (
-	// VolumeHandle is prefixed by Pod UID.
-	CSIVolumeHandlePrefixPod CSIVolumeHandlePrefix  = "Pod"
-	// VolumeHandle is prefixed by UID of the namespace where the pod is located.
-	CSIVolumeHandlePrefixNamespace CSIVolumeHandlePrefix  = "Namespace"
-	// VolumeHandle is not modified.
-	CSIVolumeHandlePrefixNone CSIVolumeHandlePrefix = "None"
-)
 ```
 
-The difference between `CSIVolumeSource` (in-lined in a pod) and `CSIPersistentVolumeSource` (in PV) are:
+CSI volume sources, that is `CSIVolumeSource` embedded in a pod specs, will be work differently than existing `CSIPersistentVolumeSource` specified in PVs:
 
 * All secret references in in-line volumes can refer only to secrets in the same namespace where the corresponding pod is running. This is common in all other volume sources that refer to secrets, incl. Flex.
-* VolumeHandle in in-line volumes can have a prefix. This prefix (Pod UID, Namespace UID or nothing) is added to the VolumeHandle before each CSI call. It makes sure that each pod uses a different volume ID for its ephemeral volumes.  The prefix must be explicitly set by pod author, there is no default.
-	* Users don't need to think about VolumeHandles used in other pods in their namespace, as each pod will get an unique prefix when `CSIVolumeHandlePrefixPod` is used. CSI volume ID with this prefix cannot accidentally conflict by another volume ID in another pod.
-	* Each pod created by ReplicaSet, StatefulSet or DaemonSet will get the same copy of a pod template. `CSIVolumeHandlePrefixPod` makes sure that each pod gets its own unique volume ID and thus can get its own volume instance.
-	* Without the prefix, user could guess volume ID of a secret-like CSI volume of another user and craft a pod with in-line volume referencing it. CSI driver, obeying idempotency, must then give the same volume to this pod. If users can use only`CSIVolumeHandlePrefixNamespace` or `CSIVolumeHandlePrefixPod`in their in-line volumes, we can make sure that they can't steal secrets of each other.
-		* `PodSecurityPolicy` will be extended to allow / deny users using in-line volumes with no prefix.
-	* Finally, `CSIVolumeHandlePrefixNone` allows selected users (based on PSP) to use persistent storage volumes in-line in pods.
+
+### Volume handles
+The VolumeHandle, in in-line volumes, will be optional for certain drivers (i.e. secrets, configMaps):
+
+* When mounting a volume and the volumeHandle is not specified (think Secrets): 
+  * Internally, CSI will check the driver's configuration object, `CSIInfo`, for the driver type.
```

Comment:
CSIInfo -> CSIDriver, see https://github.com/kubernetes/community/pull/2514

"CSI will check the driver's" - it's not CSI, it's Kubernetes / kubelet.

And it would be helpful to include here exactly what change in CSIDriver is needed. You refer to "driver type" as "ephemeral" and "remote/persistent" in the text below, but the enum values you suggest are "PodVolumeHandleGenerated" and "NoVolumeHandleGenerated". I'd suggest to use "PodVolumeHandleGenerated" in the text, with a note that it's intended to be used by ephemeral volumes.

Reply:
ok i will update.